components/details/DownloadDataLink.tsx

```diff
@@ -0,0 +1,63 @@
+import { Link, Text, Flex, VStack } from "@chakra-ui/react";
+import { memo } from "react";
+import { Entity } from "../../lib/model";
+import NextLink from "next/link";
+
+import { GrDocumentZip } from "react-icons/gr";
+
+export interface DownloadDataCSVProps {
+  entity: Entity;
+}
+
+const DownloadDataLink = ({ entity }: DownloadDataCSVProps) => {
+  const dataUrl = `${process.env.COKI_API_URL}/download/${entity.entity_type}/${entity.id}`;
```

Comment:
You might need to use `addBuildId` so that when a new version of the app is built, it doesn't fetch stale data from the API.